lib/derivations.nix

```diff
@@ -206,4 +208,38 @@ in
   optionalDrvAttr =
     cond:
     value: if cond then value else null;
+
+  /**
+    Wrap a derivation such that instantiating it produces a warning.
+
+    All attributes apart from `meta`, `name`, and `type` (which are used by
+    `nix search`) will be wrapped in `lib.warn`.
+
+    # Inputs
+
+    `msg`
+    : The warning message to emit (via `lib.warn`).
+
+    `drv`
+    : The derivation to wrap.
+
+    # Examples
+    :::{.example}
+    ## `lib.derivations.warnOnInstantiate` usage example
+
+    ```nix
+    {
+      myPackage = warnOnInstantiate "myPackage has been renamed to my-package" my-package;
+    }
+    ```
+
+    :::
+  */
+  warnOnInstantiate =
+    msg: drv:
+    let
+      drvToWrap = filterAttrs (name: value: !builtins.elem name [ "meta" "name" "type" ]) drv;
```

Comment:
```suggestion
      drvToWrap = removeAttrs [ "meta" "name" "type" ] drv;
```

Reply:
Oh looks like we're waiting for ofborg anyway.
May as well apply this then, but probably `removeAttrs` isn't in scope, so it's not just a suggestion commit.